cluster/node-pools/master-default/userdata.clc.yaml

```diff
@@ -545,6 +545,68 @@ storage:
                 value: https://identity.zalando.com/.well-known/openid-configuration
               - name: ENABLE_INTROSPECTION
                 value: "true"
+{{ if ne .Cluster.Environment "production" }}
+          - name: tokeninfo-sandbox
+            image: registry.opensource.zalan.do/foundation/platform-iam-tokeninfo:2fca26c
+            ports:
+            - containerPort: 9022
+            livenessProbe:
+              httpGet:
+                path: /health
+                port: 9022
+              periodSeconds: 3
+              failureThreshold: 2
+            readinessProbe:
+              httpGet:
+                path: /health
+                port: 9022
+              periodSeconds: 3
+              failureThreshold: 2
+            resources:
+              requests:
+                cpu: 100m
+                memory: 20Mi
+            env:
+            - name: OPENID_PROVIDER_CONFIGURATION_URL
+              value: https://sandbox.identity.zalando.com/.well-known/openid-configuration
+            - name: ENABLE_INTROSPECTION
+              value: "true"
+            - name: ISSUER
+              value: "https://sandbox.identity.zalando.com"
+            - name: LISTEN_ADDRESS
+              value: ":9022"
+          - name: skipper-tokeninfo-bridge
+            image: registry.opensource.zalan.do/pathfinder/skipper:v0.10.243
+            args:
+            - skipper
+            - -address=:9023
+            - -inline-routes
+            - |
+              health: Path("/healthz") -> inlineContent("ok") -> <shunt>;
+              q2h: Path("/oauth2/introspect") && QueryParam("token")
+                -> queryToHeader("token", "Authorization", "Bearer %s")
+                -> setRequestHeader("X-Checking-JWT", "true")
+                -> <loopback>;
+              prod: Path("/oauth2/introspect") && Header("X-Checking-JWT", "true") && JWTPayloadAllKV("iss", "https://identity.zalando.com")
+                -> dropRequestHeader("Authorization") -> dropRequestHeader("X-Checking-JWT")
+                -> "http://127.0.0.1:9021/oauth2/introspect";
+              sandbox: Path("/oauth2/introspect") && Header("X-Checking-JWT", "true") && JWTPayloadAllKV("iss", "https://sandbox.identity.zalando.com")
+                -> dropRequestHeader("Authorization") -> dropRequestHeader("X-Checking-JWT")
+                -> "http://127.0.0.1:9022/oauth2/introspect";
+              fallback: Path("/oauth2/introspect") && QueryParam("token") && Header("X-Checking-JWT", "true")
+                -> inlineContent("{\"active\":false}") -> <shunt>;
```

Comment:
`/introspect` works different than `/tokeninfo` in that it expects a token as part of the query instead of the `Authorization` header. In order to use `JWTPayloadAllKV` we need to to `queryToHeader` the token first but also leave the `token` in the query.

This can lead to an infinite route matching loop as we don't remove the `query` as seen in other examples. Adding the "X-Checking-JWT" header kind of solves it. However, we're missing the rule for when `token` isn't present at all. Currently this returns a `404` when it should result in the same as `fallback`. Adding a rule for that resulted in other rules not matching correctly anymore so I leave it like that for now.

Reply:
Readded the final rule as e2e passed before.

Reply:
Made rules more specific to avoid conflicts.